synapse/tools/apikey.py

```diff
@@ -0,0 +1,93 @@
+import sys
+import asyncio
+import argparse
+
+import synapse.exc as s_exc
+import synapse.telepath as s_telepath
+
+import synapse.lib.time as s_time
+import synapse.lib.output as s_output
+
+descr = '''
+Add, list, or delete user API keys from a Synapse service.
+'''
+
+def printkey(outp, info, apikey=None):
+    iden = info.get('iden')
+    name = info.get('name')
+    created = info.get('created')
+    updated = info.get('updated')
+    expires = info.get('expires')
+
+    outp.printf(f'Iden: {iden}')
+    if apikey:
+        outp.printf(f'  API Key: {apikey}')
+    outp.printf(f'  Name: {name}')
+    outp.printf(f'  Created: {s_time.repr(created)}')
+    outp.printf(f'  Updated: {s_time.repr(updated)}')
+    if expires:
+        outp.printf(f'  Expires: {s_time.repr(expires)}')
+
+    outp.printf('')
+
+async def main(argv, outp=s_output.stdout):
+
+    pars = argparse.ArgumentParser(prog='apikey', description=descr)
+    pars.add_argument('--svcurl', default='cell:///vertex/storage', help='The telepath URL of the Synapse service.')
+
+    subpars = pars.add_subparsers(dest='action', required=True)
+
+    addp = subpars.add_parser('add', help='Add a user API key.')
+    addp.add_argument('-d', '--duration', type=int, help='The duration of the API key in seconds.')
+    addp.add_argument('-u', '--username', type=str, help='The username to add an API key to (restricted to admins).')
+    addp.add_argument('name', help='The name of the API key to add.')
+
+    listp = subpars.add_parser('list', help='List user API keys.')
+    listp.add_argument('-u', '--username', type=str, help='The username to list API keys for (restricted to admins).')
```

Comment:
They just need the `auth.user.set.apikey` perm, not full admin though right?

Reply:
you need to be admin to resolve the username to the user iden.

Reply:
An alternative is to just use idens.